### DISCUSSION TITLE
Support for dataclass `extra="allow"`

### DISCUSSION THREAD

**Discussion by gordonhart** [pydantic/pydantic#7362] [Question]
I am curious about the current state of support for `extra="allow"` config in Pydantic dataclasses.

#2557 introduced support for extra kwargs passed to dataclass `__init__`, with the noted caveat that these extra fields are not surfaced via `__str__`. As far as I can tell, this seems to still be the behavior (see example below). However, #6470 updated the [associated documentation](https://docs.pydantic.dev/latest/usage/dataclasses/#dataclass-config) to remove copy indicating support for `extra="allow"` config, and instead state:

> Pydantic dataclasses do not support `extra='allow'`, where extra fields passed to the initializer would be stored as extra attributes on the dataclass.

This leads me to my question: what is the current state of support for `extra` config on Pydantic dataclasses? Are there additional caveats to be aware of in addition to the absence of these extra fields in `__str__`? Thanks!

### Example

Script demonstrating the current behavior:

```py
from pydantic.dataclasses import dataclass

@dataclass(config=dict(extra="allow"))
class Example:
    ...

obj = Example(a=1, b=2)
print(obj)
print(f"Extra: a={obj.a}, b={obj.b}")
```

This script yields the following output when running with the latest `pydantic==2.3.0`:

```
Example()
Extra: a=1, b=2
```

The options `forbid` and `ignore` also behave as expected:

- With `extra="forbid"`, the above script fails with `ValidationError` on instantiation of the `Example` object with extra kwargs `a` and `b`.
- With `extra="ignore"` or no `extra` config (default is `ignore`), the `Example` object is successfully instantiated, but the script fails with `AttributeError` when `obj.a` and `obj.b` are accessed.

**Comment by hramezani** [comment]
Thanks @gordonhart for finding this and creating the discussion 🙏 

Unfortunately, The documentation is wrong here, and Pydantic dataclasses supports `extra='allow'`.

We have a test for this behavior as well.
https://github.com/pydantic/pydantic/blob/a825f162daf870e210e262a403798edaf75fe4f3/tests/test_dataclasses.py#L1454-L1461

Would you like to open a PR and fix the documentation?

**Reply by gordonhart** [reply]
Opened #7375 to update the relevant documentation.

**Comment by hramezani** [comment]
Doc fixed in https://github.com/pydantic/pydantic/pull/7375 by @gordonhart
